### DISCUSSION TITLE
Transfer data to the parent form using the "InplaceAdd" instruction

### DISCUSSION THREAD

**Discussion by patriciomh1965** [serenity-is/Serenity#5729] [Help]
![InplaceAdd](https://user-images.githubusercontent.com/42218266/119044038-b39bfc80-b987-11eb-9978-49887167d89a.png)
Hello Community

I need to solve the following problem: I am using the "InplaceAdd" instruction to create a foreign entity, once the information is created I need to deliver a "CodPlanProd" data to the Parent form "Cod PlanProd". Is there an instruction to transfer the data to be displayed online?

I leave an image and the instruction of the call to the foreign entity 

XYZRow  (dad)
.......
......
......

        [DisplayName("Plan Producto"), Column("pln_plan_producto"), ForeignKey(typeof(TaCncParamPlanProductoNuevoRow)), LeftJoin("ppr"), TextualField("PprDescripcion")]
        [LookupEditor(typeof(TaCncParamPlanProductoNuevoRow), CascadeFrom = "TprNumero", CascadeField = "PprTipoProducto", InplaceAdd = true)]        
        public Int32? PprNumero
        {
            get { return Fields.PprNumero[this]; }
            set { Fields.PprNumero[this] = value; }
        }
        [DisplayName("Plan Producto")]
        [Origin("ppr"), MinSelectLevel(SelectLevel.List)]
        public String PprDescripcion
        {
            get { return Fields.PprDescripcion[this]; }
            set { Fields.PprDescripcion[this] = value; }
        }
        [DisplayName("Cod PlanProd"), Column("ppr_cod_planprod"), DefaultValue(0)]
        [Origin("ppr"), MinSelectLevel(SelectLevel.List)]
        public Int64? PprCodPlanProd 
        {
            get { return Fields.PprCodPlanProd[this]; }
            set { Fields.PprCodPlanProd[this] = value; }
        }
......
.......

XYZRow (child)
....
.....
.....
    [DisplayName("Numero"), Column("ppr_numero"), Identity]
        public Int32? PprNumero
        {
            get { return Fields.PprNumero[this]; }
            set { Fields.PprNumero[this] = value; }
        }
       
        [DisplayName("Descripcion"), Column("ppr_descripcion"), Size(100), QuickSearch, LookupInclude]
        public String PprDescripcion
        {
            get { return Fields.PprDescripcion[this]; }
            set { Fields.PprDescripcion[this] = value; }
        }

        [DisplayName("CodPlanProd"), Column("ppr_cod_planprod"),DefaultValue(0), LookupInclude]
        public Int64? PprCodPlanProd
        {
            get { return Fields.PprCodPlanProd[this]; }
            set { Fields.PprCodPlanProd[this] = value; }
        }
....
.....

I appreciate the help you can give

Atte.-

**Comment by minhhungit** [comment]
Hi @patriciomh1965 

You can handle it by implement change event of lookup editor (dropdown)

On your dialog.ts (write code somewhere ex constructor ):

```javascript
this.form.PprNumero.change(e => {
      let pprNumero = Serenity.EditorUtils.getValue(this.form.PprNumero);
      if (pprNumero != null && typeof pprNumero != 'undefined' && !Q.isEmptyOrNull(pprNumero)) {
          Q.notifySuccess(pprNumero);
         // after you have Id, you can load it via service Retrieve or loading from lookup items, then set value here using Serenity.EditorUtils for example:
        
         Serenity.EditorUtils.setValue(this.form.PprCodPlanProd, "your_Value");
      }
}
```

**Reply by patriciomh1965** [reply]
Thanks minhhungit
I am going to implement it

Just a question, this should be built in the child's dialog.ts?

**Reply by minhhungit** [reply]
main dialog

**Reply by patriciomh1965** [reply]
Hello minhhungit

I did the following programming in: 

PgmPlanesDialog ( Parent )
.....
.....
            // cambia plan producto
            this.form.PprNumero.change(() => {
                var existedItem = TaCncParamPlanProductoDatosRow.getLookup().itemById[this.form.PprNumero.value];
                if (existedItem != null && typeof (existedItem) != 'undefined') {
                    Serenity.EditorUtils.setValue(this.form.PprCodPlanProd, existedItem.PprCodPlanProd);                   
                }
            });
....
.....


The instruction applied to the child is the following:

TaCncParamPlanProductoNuevoDialog ( Child )

namespace PLAINT.ServiciosProducto {

    @Serenity.Decorators.registerClass()   
    export class TaCncParamPlanProductoNuevoDialog extends Serenity.EntityDialog<TaCncParamPlanProductoNuevoRow, any> {
        protected getFormKey() { return TaCncParamPlanProductoNuevoForm.formKey; }
        protected getIdProperty() { return TaCncParamPlanProductoNuevoRow.idProperty; }
        protected getLocalTextPrefix() { return TaCncParamPlanProductoNuevoRow.localTextPrefix; }
        protected getNameProperty() { return TaCncParamPlanProductoNuevoRow.nameProperty; }
        protected getService() { return TaCncParamPlanProductoNuevoService.baseUrl; }
                
        protected form = new TaCncParamPlanProductoNuevoForm(this.idPrefix);
        
         constructor() {
            super();

            this.form = new TaCncParamPlanProductoNuevoForm(this.idPrefix);
            Serenity.SubDialogHelper.triggerDataChange(this);

            this.form.PprTipoProducto.change(() => {
                // codigo lk nuevo
                if (this.isNew()) {
                    // trae el valor maximo
                    Q.serviceRequest<any>('ServiciosProducto/TaCncParamPlanProductoNuevo/Maximo',
                        {
                            TipoValue: 4,
                            CodigoValue: this.form.PprTipoProducto.value // tipo producto
                        }, response => {
                            this.form.PprCodPlanProd.value = response.MaximoValue;                           
                        });
                }
            });
        }

        protected afterLoadEntity() {
            super.afterLoadEntity();
            Serenity.EditorUtils.setReadOnly(this.form.PprCodPlanProd, true);
            if (this.isNew()) {
                Q.serviceRequest<any>('ServiciosProducto/TaCncParamPlanProductoNuevo/Maximo',
                    {
                        TipoValue: 4,
                        CodigoValue: this.form.PprTipoProducto.value // tipo producto
                    }, response => {
                        this.form.PprCodPlanProd.value = response.MaximoValue;
                    });
            }
        }
    }
}


When a new record (child) is created, the new data "PprCodPlanProd" cannot be displayed in the Parent form.
Dear, I think I am missing something that I cannot find.

What you can tell me helps me.

Atte.
